hms-api/__tests__/service/idea-comment-service.test.ts

```diff
@@ -0,0 +1,147 @@
+import * as userRepository from '../../src/repository/user-repository';
+import * as ideaCommentRepository from '../../src/repository/idea-comment-repository';
+import * as ideaRepository from '../../src/repository/idea-repository';
+import {
+  createIdeaComment,
+  getIdeaCommentListResponse,
+  getIdeaCommentResponse,
+} from '../../src/service/idea-comment-service';
+import {uuid} from '../../src/util/Uuid';
+import ReferenceNotFoundError from '../../src/error/ReferenceNotFoundError';
+import {randomIdea} from '../repository/domain/idea-maker';
+import {
+  IdeaCommentData,
+  makeIdeaComment,
+  randomIdeaComment,
+} from '../repository/domain/ideaComment-maker';
+import IdeaCommentResponse from '../../src/rest/ideaComment/IdeaCommentResponse';
+import {randomUser} from '../repository/domain/user-maker';
+import ValidationError from '../../src/error/ValidationError';
+import IdeaCommentListResponse from '../../src/rest/ideaComment/IdeaCommentListResponse';
+import NotFoundError from '../../src/error/NotFoundError';
+
+const mockPutIdeaComment = jest.fn();
+jest
+  .spyOn(ideaCommentRepository, 'putIdeaComment')
+  .mockImplementation(mockPutIdeaComment);
+const mockUserExists = jest.fn();
+jest.spyOn(userRepository, 'userExists').mockImplementation(mockUserExists);
+const mockIdeaExists = jest.fn();
+jest.spyOn(ideaRepository, 'ideaExists').mockImplementation(mockIdeaExists);
+const mockGetUser = jest.fn();
+jest.spyOn(userRepository, 'getUser').mockImplementation(mockGetUser);
+const mockGetIdea = jest.fn();
+jest.spyOn(ideaRepository, 'getIdea').mockImplementation(mockGetIdea);
+const mockGetIdeaComment = jest.fn();
+jest
+  .spyOn(ideaCommentRepository, 'getIdeaComment')
+  .mockImplementation(mockGetIdeaComment);
+const mockGetUsers = jest.fn();
+jest.spyOn(userRepository, 'getUsers').mockImplementation(mockGetUsers);
+const mockListIdeaComments = jest.fn();
+jest
+  .spyOn(ideaCommentRepository, 'listIdeaComments')
+  .mockImplementation(mockListIdeaComments);
+
+describe('Create Idea Comment', () => {
+  test('Happy Path', async () => {
+    mockUserExists.mockResolvedValue(true);
+    mockIdeaExists.mockResolvedValue(true);
+
+    const expected = randomIdeaComment();
+
+    expect(
+      await createIdeaComment(expected.ideaId, expected.userId, expected.text),
```

Comment:
so since there are two versions of this function, one with the parentCommentId and one without you should really have two tests for it